### PR TITLE
fix partition creation on historical data

### PR DESCRIPTION
**Pull request by dinal**

### CODE REVIEW

File: pkg/partmgr/partmgr.go [hunk 1]
```diff
@@ -157,8 +161,19 @@ func (p *PartitionManager) createNewPartition(t int64) (*DBPartition, error) {
 		return nil, err
 	}
 	p.currentPartitionInterval = partition.partitionInterval
```
Comment (imakhlin) [inline pkg/partmgr/partmgr.go]: It would be better to separate creation logic from update current partition details

File: pkg/partmgr/partmgr.go [hunk 2]
```diff
@@ -144,9 +144,13 @@ func (p *PartitionManager) TimeToPart(t int64) (*DBPartition, error) {
 					return p.partitions[i], nil
 				}
 			}
+			head := p.headPartition
+			part, _ := p.createNewPartition(p.currentPartitionInterval * (t / p.currentPartitionInterval))
```
Comment (imakhlin) [inline pkg/partmgr/partmgr.go]: if you'll separate `create` and `set` your code will be mush more readable

File: pkg/partmgr/partmgr.go [hunk 3]
```diff
@@ -169,7 +184,9 @@ func (p *PartitionManager) updatePartitionInSchema(partition *DBPartition) error
 	if err != nil {
 		return errors.Wrap(err, "Failed to update new partition in schema file")
 	}
-	err = p.container.Sync.PutObject(&v3io.PutObjectInput{Path: path.Join(p.path, config.SCHEMA_CONFIG), Body: data})
+	if p.container != nil {
+		err = p.container.Sync.PutObject(&v3io.PutObjectInput{Path: path.Join(p.path, config.SCHEMA_CONFIG), Body: data})
+	}
```
Comment (imakhlin) [inline pkg/partmgr/partmgr.go]: else what?

File: pkg/partmgr/partmgr_test.go [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+	part, _ = manager.TimeToPart(startTime + (interval * 2))
+	assert.Equal(tst, 3, len(manager.partitions))
+	assert.Equal(tst, manager.partitions[1], part)
+	//add first
+	part, _ = manager.TimeToPart(startTime)
+	assert.Equal(tst, 4, len(manager.partitions))
+	assert.Equal(tst, manager.partitions[0], part)
+}
+
+func createSchema() *config.Schema {
```
Comment (imakhlin) [inline pkg/partmgr/partmgr_test.go]: reuse or modify  pkg/tsdb/tsdbtest/tsdbtest.go:CreateSchema

Comment (dinal) [reply pkg/partmgr/partmgr_test.go]: i can't, it complainד about cyclic import

File: pkg/partmgr/partmgr.go (hunk 3 above)
Comment (dinal) [reply pkg/partmgr/partmgr.go]: else nothing, file will not be updated (it can be nil only in tests)

Comment (imakhlin) [reply pkg/partmgr/partmgr.go]: it seems to me that `nil` container is an error rather than valid state

File: pkg/partmgr/partmgr_test.go (hunk 4 above)
Comment (imakhlin) [reply pkg/partmgr/partmgr_test.go]: this is no excuse for code duplication - please refactor if necessary

**Pull request merged**